### DISCUSSION TITLE
Server Push API

### DISCUSSION THREAD

**Discussion by EugenEistrach** [ShafSpecs/remix-pwa#33] [Q&A] [answered]
Hello, 
was reading the docs and am am wondering if its possible to push notification from server to specific users? 
The example in the docs looks like it will broadcast the notification to everybody. 

If I somebody writes a message to another person I want to notify the other person about that message. Is this possible?

**Comment by moishinetzer** [comment]
Any updates on this? I'm also curious about this.

**Comment by ShafSpecs** [comment]
Do you mean like a messaging app? I believe this should be possible with push notifications + sockets

I did read something on this before and Notifications API could work too, you would have a listener on the client that checks if a message has come in and trigger the notification. Or you could also do it server-side. This is an interesting one

**Reply by ShafSpecs** [reply]
I am coming back to maintain this project actively so I would be exploring more possibilities and also updating the docs accordingly

**Comment by ShafSpecs** [comment, marked as answer]
@EugenEistrach @moishinetzer 
https://remix-pwa-docs.vercel.app/push/chat-push

Very possible now, with remix-pwa broken down into modules, I released a guide on how to accomplish just that! 👍
